src/main/java/bio/terra/service/dataset/flight/create/CreateDatasetAuthzResource.java

```diff
@@ -44,6 +50,9 @@ public StepResult doStep(FlightContext context) {
         Dataset dataset = datasetService.retrieve(datasetId);
         List<String> policyEmails = sam.createDatasetResource(userReq, datasetId);
         bigQueryPdao.grantReadAccessToDataset(dataset, policyEmails);
+        DatasetModel datasetModel = datasetService.retrieveModel(datasetId);
```

Comment:
I think this retrieveModel call causes a duplicate call to retrieve above. If you just need the data project, perhaps you could just call datalocationservice instead?

Reply:
i looked into it, and the retrieveModel is making a duplicate call -- i just added another retreiveModel that takes in a dataset which skips the duplicate call and just deals with the DataLocationService :)